src/superqt/utils/_animation.py

```diff
@@ -0,0 +1,76 @@
+"""Different animators to be used for animating components"""
+# from math import sin, cos, radians
+
+from ..qtcompat.QtCore import QEasingCurve, QPropertyAnimation, QVariantAnimation
+from ..qtcompat.QtGui import QIcon, QPixmap, QTransform
+from ..qtcompat.QtWidgets import QAbstractButton, QWidget
+
+
+# =================================================================================================
+def create_hide_show_animation(
+    widget: QWidget = None,
+    duration: int = 500,
+    easing_curve: QEasingCurve = QEasingCurve.Type.InOutCubic,
+) -> QPropertyAnimation:
+    """
+    Creates an animation that can be used to show animation while hiding and revealing the widget.
+    """
+    animation: QPropertyAnimation
+    if widget is not None:
+        animation = QPropertyAnimation(
+            targetObject=widget, propertyName=b"maximumHeight"
+        )
+        animation.setEndValue(widget.sizeHint().height() + 10)
+    else:
+        animation = QPropertyAnimation()
+        animation.setEndValue(0)
+
+    # Create the animation
+    animation.setStartValue(0)
+    animation.setEasingCurve(easing_curve)
+    animation.setDuration(duration)
+
+    return animation
+
+
+# =================================================================================================
+def create_icon_rotation_animation(
```

Comment:
same here, make private in the other module

Reply:
migrated.